src/llm/legacy/servable.cpp

```diff
@@ -0,0 +1,158 @@
+    SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Complete unary response: {}", executionContext->response);
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::preparePartialResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    std::stringstream ss;
+    ss << executionContext->lastStreamerCallbackOutput;
+    std::string lastTextChunk = ss.str();
+    executionContext->lastStreamerCallbackOutput = "";
+    auto generationStatus = legacyExecutionContext->finished.wait_for(std::chrono::nanoseconds::zero());
+    if (generationStatus != std::future_status::ready) {  // continue
+        if (lastTextChunk.size() > 0) {
+            executionContext->response = wrapTextInServerSideEventMessage(executionContext->apiHandler->serializeStreamingChunk(lastTextChunk, ov::genai::GenerationFinishReason::NONE));
+            SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Generated subsequent streaming response: {}", executionContext->response);
+        }
+        executionContext->sendLoopbackSignal = true;
+    } else {  // finish generation
+        OVMS_PROFILE_SCOPE("Generation of last streaming response");
+        executionContext->textStreamer->end();
+        // if streamer::put returned a value, streamer::end() result will not contain it, so we add it manually
+        if (!executionContext->lastStreamerCallbackOutput.empty())
+            lastTextChunk = lastTextChunk + executionContext->lastStreamerCallbackOutput;
+        executionContext->response = wrapTextInServerSideEventMessage(executionContext->apiHandler->serializeStreamingChunk(lastTextChunk, ov::genai::GenerationFinishReason::STOP));
+        if (executionContext->apiHandler->getStreamOptions().includeUsage)
+            executionContext->response += wrapTextInServerSideEventMessage(executionContext->apiHandler->serializeStreamingUsageChunk());
+
+        executionContext->response += wrapTextInServerSideEventMessage("[DONE]");
+
+        SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Generated complete streaming response: {}", lastTextChunk);
+        executionContext->sendLoopbackSignal = false;
+        return absl::OkStatus();
+    }
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::readPartialExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Relevant properties read");
+    return absl::OkStatus();
+}
+
+bool LegacyServable::supportsSpeculativeDecoding() const {
+    return true;
```

Comment:
Not sure if we should set it to true before testing proves it actually works as expected.